src/java/org/apache/cassandra/repair/ValidationTask.java

```diff
@@ -70,56 +72,50 @@ public synchronized void treesReceived(MerkleTrees trees)
     {
         if (trees == null)
         {
-            active = false;
             tryFailure(RepairException.warn(desc, previewKind, "Validation failed in " + endpoint));
         }
-        else if (active)
-        {
-            trySuccess(new TreeResponse(endpoint, trees));
-        }
-        else
+        else if (!trySuccess(new TreeResponse(endpoint, trees)))
         {
-            // If the task has already been aborted, just release the possibly off-heap trees and move along.
+            // If the task is done, just release the possibly off-heap trees and move along.
             trees.release();
-            trySuccess(null);
         }
     }
 
+    public synchronized void abort()
+    {
+        abort(new RuntimeException("TODO"));
```

Comment:
nit: Cleanup reminder

Would an unchecked `RepairAbortedException` be useful...or is it a waste, given we'd still have basically everything we need in the exception message anyway?

Reply:
removed, I left it there while refactoring... no longer used

> Would an unchecked RepairAbortedException be useful...or is it a waste, given we'd still have basically everything we need in the exception message anyway?

this was a tmp method while I was refactoring to let us cleanup properly... we now use the actual error and not a random one